src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -262,13 +263,21 @@ func (sms *StorageMiningSubsystem_I) VerifyElectionPoSt(header block.BlockHeader
 		return false
 	}
 
-	pow, err := sma._getActivePower()
-	if err != nil {
+	activeSectorWeight, inactiveSectorWeight, ok := spa.GetSectorWeightForMiner(header.Miner())
+	if !ok {
+		// TODO: better error handling
+		return false
+	}
+
+	currPledge, ok := spa.GetCurrPledgeForMiner(header.Miner())
+	if !ok {
 		// TODO: better error handling
 		return false
 	}
 
-	if !spa.ActivePowerMeetsConsensusMinimum(pow) {
+	pow := inds.StoragePower(activeSectorWeight, inactiveSectorWeight, currPledge)
```

Comment:
[block] odd to have StoragePower fn be in the inds. It should be a part of the SPA..?

Reply:
understood, it depends on how we set StoragePower, we can potentially pull out the logic from indices to SPA when dust settles but if the computation involves macroeconomic indicators, indices will still be involved.

Reply:
Sure, but pull indices into SPA with runtime. This breaks abstraction barriers